zappa/core.py

```diff
@@ -1366,6 +1369,9 @@ def update_lambda_configuration(
         if not layers:
             layers = []
 
+        # Wait until function is ready, otherwise expected keys will be missing from 'lambda_aws_config'.
+        self.wait_until_lambda_function_is_updated(function_name)
```

Comment:
Thinking about this more, I wonder if this might better live at the end of `update_lambda_function` just like the other waiting function is living at the end of `create_lambda_function` - that might be a safer way to ensure we are waiting for the function to be ready in all cases after create/update.  LGTMO